includes/Core/Authentication/Authentication.php

```diff
@@ -1386,6 +1389,29 @@ private function filter_features_via_proxy( $feature_enabled, $feature_name ) {
 			return $feature_enabled;
 		}
 
+		// The experimental features (ideaHubModule and swgModule) are checked within Modules::construct() which
+		// runs before Modules::register() where the `googlesitekit_features_request_data` filter is registered.
+		// Without this filter, some necessary context data is not sent when a request to Google_Proxy::get_features()
+		// is made. So we skip making this request and solely check the database to see if these features are enabled.
```

Comment:
Maybe worth briefly explaining the `$this->transients->get()` call in the code here as well: Since a request is only being made if the transient isn't set, we only need to avoid _that_ scenario. If the data is already cached in a transient, we still need to use the data from that, i.e. fall through to `$this->get_transient_features()`.